app/helpers/pieces_helper.rb

```diff
@@ -0,0 +1,12 @@
+module PiecesHelper
+  def render_piece_highlight(x, y)
+    piece = @game.retrieve_piece(x, y)
+    if !piece.nil?
```

Comment:
I didn't write this code, I know what you are saying about "if/not" it's confusing - but I don't know where you are getting -
`return unless piece.present?`
I don't see that in the file.